File: recipe-sharing-client/src/services/recipes/RecipeDto.ts
import IngredientDTO from "../ingredients/IngredientDto";
import InstructionDTO from "../instructions/InstructionDto";

export default interface RecipeDTO {
    id: number;
    name: string;
    description: string;
    ingredients: IngredientDTO[];
    instructions: InstructionDTO[];
    minutes: number;
    servings: number;
    price: number;
    link: string;
    imageUrl: string;
    tags: string[];
}

File: recipe-sharing-api/src/main/java/com/alechoskins/RecipeSharingApi/dto/recipe/RecipeDto.java
package com.alechoskins.RecipeSharingApi.dto.recipe;

import com.alechoskins.RecipeSharingApi.dto.recipe_ingredients.Recipe_IngredientDto;
import java.util.List;

public class RecipeDto {

    //region PROPERTIES
    private Long id;
    private String name;
    private String description;
    private int minutes;
    private double servings;
    private Double lastRecordedPrice;
    private String link;
    private String imageUrl;
    //#endregion

    //#region NAV PROPERTIES
    private List<Recipe_IngredientDto> recipeIngredients;
    //#endregion

}
